bionic/core/flow_execution.py

```diff
@@ -407,47 +410,58 @@ def _mark_entry_active(self, entry):
 
     def _wait_on_in_progress_entries(self):
         "Waits on any in-progress entry to finish."
-        futures = set(entry.future for entry in self._in_progress_entries)
-        finished_futures, _ = wait(futures, return_when=FIRST_COMPLETED)
-        try:
-            for finished_future in finished_futures:
-                task_keys = finished_future.result()
-                for task_key in task_keys:
-                    entry = self._entries_by_task_key[task_key]
-                    entry.state.sync_after_remote_computation()
-                    assert self._mark_entry_completed_if_possible(entry)
-        except Exception as exception:
-            # If there is an error, wait until all futures are done. With AIP
-            # execution, we can have one task fail while others are running. If
-            # we don't wait till those tasks are done, the user might perform
-            # another `get` operation that can spawn the same Task again.
-
-            logger.error(
-                oneline(
-                    f"""
-                    Encountered an error while doing remote computation:
-                    {exception!r}. Waiting for all other remote computation to
-                    complete...
-                    """
-                )
-            )
-            finished_futures, _ = wait(futures, return_when=ALL_COMPLETED)
-
-            # There may be errors in the other tasks. Ensure that we log all of
-            # the other errors as well.
-            for future in finished_futures:
-                e = future.exception()
-                if e != exception and e is not None:
-                    logger.error(
-                        oneline(
-                            f"""
-                        Encountered another exception while doing remote
-                        computation: {e!r}.
+        entries_by_future = {entry.future: entry for entry in self._in_progress_entries}
+        finished_futures, _ = wait(
+            entries_by_future.keys(), return_when=FIRST_COMPLETED
+        )
+        for finished_future in finished_futures:
+            try:
+                self._sync_remotely_computed_task_keys(finished_future.result())
```

Comment:
It might be good to include the word `complete` in this method name -- the fact that we are transitioning the entries from `IN_PROGRESS` to `COMPLETED` is important.

Reply:
I can't think of a good way to rename this method without having a very long name. E.g. `_sync_remotely_computed_task_keys_and_mark_complete_if_possible`. or `_sync_remotely_computed_and_mark_complete_if_possible`

Reply:
In this case, it will always be possible to mark them complete, right? So we could write `_sync_and_complete_remotely_computed_task_keys`. ("mark completed" would be more precise but this seems like an okay compromise to me.)